spec/concepts/tribunal_commerce/stock/operation/import_spec.rb

```diff
@@ -0,0 +1,110 @@
+require 'rails_helper'
+
+RSpec::Matchers.define_negated_matcher :not_change, :change
+
+describe TribunalCommerce::Stock::Operation::Import do
+  let(:stock_args) { { year: '2017', month: '01', day: '28' } }
+  let(:logger) { spy(Rails.logger) }
+
+  subject { described_class.call(stock_args: stock_args, stocks_folder: fixture_path, logger: logger) }
+
+  it 'logs the operation start fetching the specified stock' do
+    stock_name = "#{stock_args[:year]}/#{stock_args[:month]}/#{stock_args[:day]}"
+    subject
+
+    expect(logger).to have_received(:info)
+      .with("Starting import : looking for stock #{stock_name} in file system...")
+  end
+
+  # Fixture stock folder contains three units :
+  # spec/fixtures/tc/stock/2017/01/28
+  # ├── 0121_S1_20170128.zip
+  # ├── 1001_S1_20170128.zip
+  # └── 5432_S1_20170128.zip
+  context 'when the specified stock is found in the file system' do
+    let(:fixture_path) { Rails.root.join('spec/fixtures/tc/stock') }
+    let(:new_stock) { subject[:new_stock] }
+
+    it { is_expected.to be_success }
+
+    it 'drops db indexes' do
+      expect_any_instance_of(ActiveRecord::ConnectionAdapters::PostgreSQLAdapter)
+        .to receive(:execute)
+        .exactly(expected_drop_query_count).times
+      subject
+    end
+
+    it 'creates a job to import each stock unit' do
+      stock_unit_ids = new_stock.stock_units.pluck(:id)
+
+      stock_unit_ids.each do |id|
+        expect(ImportTcStockUnitJob)
+          .to have_been_enqueued.with(id).on_queue("rncs_worker_api_entreprise_#{Rails.env}_tc_stock")
+      end
+    end
+
+    it 'creates a StockTribunalCommerce with status "PENDING"' do
```

Comment:
le nom du test parle du status "PENDING" mais ça n'est pas testé.
Soit `it 'creates a valid Stock...`
ou tester que  le status est bien à PENDING (qui est indirectement testé à la spec suivante)

Reply:
Un oubli, c'est corrigé.